benchmarks/kokkos/main.cpp

```diff
@@ -0,0 +1,626 @@
+
+    Kokkos::parallel_for(10, KOKKOS_LAMBDA(int i) {
+        auto random_generator = random_pool.get_state();
+        targets(i, 0) = Kokkos::rand<decltype(random_generator), UINT>::draw(random_generator, 0, n_qubits - 2);
+        targets(i, 1) = Kokkos::rand<decltype(random_generator), UINT>::draw(random_generator, 0, n_qubits - 1);
+        if(targets(i, 0) == targets(i, 1)) targets(i, 1) = n_qubits - 1;
+        random_pool.free_state(random_generator);
+    });
+
+    Kokkos::MDRangePolicy<Kokkos::Rank<3>> policy({0, 0, 0}, {10, 4, 4});
+    Kokkos::parallel_for(policy, KOKKOS_LAMBDA(int i, int j, int k) {
+        auto random_generator = random_pool.get_state();
+        double real_part = Kokkos::rand<decltype(random_generator), double>::draw(random_generator);
+        double imag_part = Kokkos::rand<decltype(random_generator), double>::draw(random_generator);
+        matrixes(i, j, k) = CTYPE(real_part, imag_part);
+        random_pool.free_state(random_generator);
+    });
+        
+    auto start_time = std::chrono::high_resolution_clock::now();
+    Kokkos::View<UINT*> target("target", 2);
+    Kokkos::View<CTYPE**> matrix("matrix", 4, 4);
+    for (int i = 0; i < 10; ++i) {
+        Kokkos::deep_copy(target, Kokkos::subview(targets, i, Kokkos::ALL));
+        Kokkos::deep_copy(matrix, Kokkos::subview(matrixes, i, Kokkos::ALL, Kokkos::ALL));
+        update_with_dense_matrix(state, n_qubits, target, matrix);
+    }
+    Kokkos::fence();
+    auto end_time = std::chrono::high_resolution_clock::now();
+    return std::chrono::duration_cast<std::chrono::nanoseconds>(end_time - start_time).count();
+}
+
+double double_control_matrix_bench(UINT n_qubits) {
+    assert(n_qubits >= 3);
+    Kokkos::View<UINT**> targets("targets", 10, 1);
+    Kokkos::View<UINT**> control_list("control_list", 10, 2);
+    Kokkos::View<UINT**> control_values("control_values", 10, 2);
+    Kokkos::View<CTYPE***> matrixes("matrixes", 10, 2, 2);
+    auto state(make_random_state(n_qubits));
+    Kokkos::fence();
+
+    unsigned seed = std::chrono::high_resolution_clock::now().time_since_epoch().count();
+    Kokkos::Random_XorShift64_Pool<> random_pool(seed);
+
+    Kokkos::parallel_for(10, KOKKOS_LAMBDA(int i) {
+        auto random_generator = random_pool.get_state();
+        targets(i, 0) = Kokkos::rand<decltype(random_generator), UINT>::draw(random_generator, 0, n_qubits - 1);
+        control_list(i, 0) = Kokkos::rand<decltype(random_generator), UINT>::draw(random_generator, 0, n_qubits - 1);
+        control_list(i, 1) = Kokkos::rand<decltype(random_generator), UINT>::draw(random_generator, 0, n_qubits - 2);
+        if(targets(i, 0) == control_list(i, 0)) control_list(i, 0) = n_qubits - 1;
+        if(targets(i, 0) == control_list(i, 1)) control_list(i, 1) = n_qubits - 2;
+        if(control_list(i, 0) == control_list(i, 1)) {
+            if(n_qubits - 1 == targets(i, 0)) control_lists(i, 1) = n_qubits - 2;
```

Comment:
`control_lists` が未定義でビルドできなかったので確認お願いします！
```suggestion
            if(n_qubits - 1 == targets(i, 0)) control_list(i, 1) = n_qubits - 2;
```

Reply:
修正しました！すみません！